iroha/src/asset.rs

```diff
@@ -6,101 +6,103 @@ use parity_scale_codec::{Decode, Encode};
 pub struct Asset {
     /// identifier of asset, formatted as asset_name#domain_id
     pub id: Id,
-    pub amount: u128,
+    pub quantity: u128,
 }
 
 impl Asset {
     pub fn new(id: Id) -> Self {
-        Asset { id, amount: 0 }
+        Asset { id, quantity: 0 }
     }
 
-    pub fn with_amount(mut self, amount: u128) -> Self {
-        self.amount = amount;
+    pub fn with_quantity(mut self, quantity: u128) -> Self {
+        self.quantity = quantity;
         self
     }
 }
 
-pub mod isi {
-    use super::*;
-    use crate::isi::Contract;
-    use iroha_derive::{IntoContract, Io};
-    use parity_scale_codec::{Decode, Encode};
+/// Identification of an Asset. Consists of Asset's name, Domain's name and Account's name.
+///
+/// # Example
+///
+/// ```
+/// use iroha::asset::Id;
+///
+/// let id = Id::new("xor", "user", "company");
+/// ```
+#[derive(Clone, Debug, PartialEq, PartialOrd, Ord, Eq, std::hash::Hash, Encode, Decode)]
+pub struct Id {
+    pub name: String,
+    pub container: String,
```

Comment:
Assets are namespaced by domain, like sora#xor (domain#asset_name). This guarantees global uniqueness of an asset identity. So I would just call this domain to be obvious to the reader.

Reply:
https://soramitsu.atlassian.net/browse/RH2-88?atlOrigin=eyJpIjoiMWY2NmQ2NDVhNjc3NGE5NWIyNGQ2MDY3NTIwOTQwOTciLCJwIjoiaiJ9